torchgeo/datasets/western_usa_live_fuel_moisture.py

```diff
@@ -0,0 +1,357 @@
+        api_key: Optional[str] = None,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new Tropical Cyclone Wind Estimation Competition Dataset.
+
+        Args:
+            root: root directory where dataset can be found
+            input_features: which input features to include
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            api_key: a RadiantEarth MLHub API key to use for downloading the dataset
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            AssertionError: if ``split`` argument is invalid
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+            RuntimeError: if ``input_features`` contains invalid variable names
+            ImportError: if pandas is not installed
+        """
+        super().__init__()
+
+        self.root = root
+        self.transforms = transforms
+        self.checksum = checksum
+        self.download = download
+        self.api_key = api_key
+
+        self._verify()
+
+        try:
+            import pandas as pd  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "pandas is not installed and is required to use this dataset"
+            )
+
+        assert all(
+            input in self.all_variable_names for input in input_features
+        ), "Invalid input variable name."
+        self.input_features = input_features
+
+        self.collection = self._retrieve_collection()
+
+        self.dataframe = self._load_data()
+        import pdb
+
+        pdb.set_trace()
+        print(0)
+
+    def _retrieve_collection(self) -> list[str]:
+        """Retrieve dataset collection that maps samples to paths.
+
+        Returns:
+            list of sample paths
+        """
+        return glob.glob(
+            os.path.join(self.root, self.collection_id, "**", "labels.geojson")
+        )
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.dataframe)
+
+    def __getitem__(self, index: int) -> dict[str, Any]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            input features and target at that index
+        """
+        data = self.dataframe.iloc[index, :]
+
+        sample: dict[str, Tensor] = {
+            "input": torch.tensor(data.drop([self.label_name]), dtype=torch.float32),
+            "label": torch.tensor(data[self.label_name], dtype=torch.float32),
+        }
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def _load_data(self):
```

Comment:
I am not sure how to handle the return type here, with the conditional pandas import.

Reply:
Try this:
```suggestion
    def _load_data(self) -> "pd.DataFrame":
```
The quotes prevent it from being evaluated at run-time.

Reply:
I still get flake8:
`torchgeo/datasets/western_usa_live_fuel_moisture.py:289:29: F821 undefined name 'pd'` 

and 

mypy:
`torchgeo/datasets/western_usa_live_fuel_moisture.py:289: error: Name "pd" is not defined  [name-defined]`

Reply:
What about `"pandas.DataFrame"`? We had to hack things to get pyvista working in idtrees.py and ended up using:
```
# type: ignore[name-defined] # noqa: F821
```
to ignore the issue. At some point in the future I may make pandas a required dep so the issue will go away.